packages/react/src/ReactLazy.js

```diff
@@ -52,14 +53,18 @@ function lazyInitializer<T>(payload: Payload<T>): T {
   if (payload._status === Uninitialized) {
     const ctor = payload._result;
     const thenable = ctor();
+    invariant(
+      thenable !== undefined && typeof thenable.then === 'function',
+      'lazy: Expects to receive an async function.',
+    );
     // Transition to the next state.
     const pending: PendingPayload = (payload: any);
     pending._status = Pending;
     pending._result = thenable;
     thenable.then(
       moduleObject => {
         if (payload._status === Pending) {
-          const defaultExport = moduleObject.default;
+          const defaultExport = moduleObject && moduleObject.default;
```

Comment:
Similarly, why isn't this property access throwing a proper error that then is passed to the error handler?

Reply:
Should I use 'invariant' for the check?

Reply:
If throwing error (invariant or other) in thenable.then, it error to be UnhandledPromiseRejectionWarning.
I think that it is a better solution than before the fix.